controller/Controller.h

```diff
@@ -246,18 +246,18 @@ std::shared_ptr<org::apache::nifi::minifi::core::controller::ControllerService>
 
   configuration->get(org::apache::nifi::minifi::Configure::nifi_provenance_repository_class_name, prov_repo_class);
   // Create repos for flow record and provenance
-  const auto prov_repo = org::apache::nifi::minifi::core::createRepository(prov_repo_class, true, "provenance");
+  const std::shared_ptr<org::apache::nifi::minifi::core::Repository> prov_repo = org::apache::nifi::minifi::core::createRepository(prov_repo_class, true, "provenance");
```

Comment:
Is there a specific reason for this change, or is it just personal preference? I would normally do the opposite change.

Reply:
For example usage at line 274 expects shared_ptr.

Reply:
So is this an invocation of the unique_ptr -> shared_ptr constuctor? In that case, maybe just saying `std::shared_ptr` (and relying on CTAD) would be enough.
```suggestion
  const std::shared_ptr prov_repo = org::apache::nifi::minifi::core::createRepository(prov_repo_class, true, "provenance");
```

Reply:
Yes, it is that way. I'll do as suggested.